src/crypto/auth.sagas.test.ts

```diff
@@ -1,18 +1,111 @@
+import { push } from 'connected-react-router';
 import keytar from 'keytar';
 import { expectSaga } from 'redux-saga-test-plan';
 import { call } from 'redux-saga-test-plan/matchers';
 
+import {
+  createPassword,
+  createPasswordSuccess,
+  login,
+  loginFailed,
+  loginSuccess,
+  rehydrateAllState,
+  resetSettings,
+  setNewUser
+} from '@common/store';
 import { KEYTAR_SERVICE } from '@config';
 import { fAccount } from '@fixtures';
 
-import { resetWorker } from './auth.sagas';
+import { ROUTE_PATHS } from '../app/routing';
+import { translateRaw } from '../common/translate';
+import {
+  checkNewUserWorker,
+  createPasswordWorker,
+  loginWorker,
+  logoutWorker,
+  resetWorker
+} from './auth.sagas';
+import {
+  checkSettingsKey,
+  clearEncryptionKey,
+  getSettingsKey,
+  hasSettingsKey,
+  init
+} from './secrets';
 
 jest.mock('keytar');
 
+describe('checkNewUserWorker', () => {
+  it('checks if the settings key exists and dispatches setNewUser', async () => {
+    await expectSaga(checkNewUserWorker)
+      .provide([[call.fn(hasSettingsKey), true]])
+      .put(setNewUser(false))
+      .silentRun();
+
+    await expectSaga(checkNewUserWorker)
+      .provide([[call.fn(hasSettingsKey), false]])
+      .put(setNewUser(true))
+      .silentRun();
+  });
+});
+
+describe('loginWorker', () => {
+  it('sets the encryption key and logs in', async () => {
+    await expectSaga(loginWorker, login('foo'))
+      .provide([[call.fn(init), undefined]])
+      .provide([[call.fn(checkSettingsKey), true]])
+      .call(init, 'foo')
+      .call(checkSettingsKey)
+      .put(loginSuccess())
+      .put(rehydrateAllState())
+      .silentRun(5000);
+  });
+
+  it('puts an error if the settings key is invalid', async () => {
+    await expectSaga(loginWorker, login('foo'))
+      .provide([[call.fn(init), undefined]])
+      .provide([[call.fn(checkSettingsKey), false]])
+      .call(init, 'foo')
+      .call(checkSettingsKey)
+      .put(loginFailed(translateRaw('LOGIN_ERROR')))
+      .silentRun(5000);
```

Comment:
Why the need for the `5000` param?

Reply:
For some reason the saga would otherwise time out

Reply:
Are we missing a mock?

Reply:
Every function that is called by the saga is mocked using `provide`, so I'm not sure why it times out